### DISCUSSION TITLE
path completion using ripgrep

### DISCUSSION THREAD

**Discussion by radiodee1** [ibhagwan/fzf-lua#1274] [General]
Hi, I have this use case where I want to search for a path, and use ripgrep. I'd like to be able to hit a keymap combination and see the rg boxes on the screen and use rg to find the file I'm looking for, and then when I select it, instead of opening that file, having the path for the file inserted at the cursor.

I feel this might be useful when searching for things like hashtags in documents.

Is this possible?

**Comment by ibhagwan** [comment]
https://github.com/ibhagwan/fzf-lua?tab=readme-ov-file#insert-mode-completion

Set cursor on any path-like item and `:FzfLua complete_path`, for insert mode use keymaps as explained above.

**Comment by radiodee1** [comment]
Just to start with, I think this is a great plugin. The complete_path function does not use the ripgrep style search mechanism. If I use complete_file with grep style searching, when I finally select something, the file gets loaded into a buffer and I can edit it. That's not what I want. When I select something I just want the path string inserted in the first document. I hope this makes sense.

**Reply by ibhagwan** [reply]
For that you’d need to implement custom completion using the `complete` function:
https://github.com/ibhagwan/fzf-lua?tab=readme-ov-file#custom-completion

With custom completion any picker can complete anything.

**Comment by radiodee1** [comment]
Hi again. I'm really not much of a lua coder. This is what I've got so far.

```lua
vim.keymap.set({ "i" }, "<C-x><C-f>",
  function()
    require("fzf-lua").fzf_exec(
        function()
            require("fzf-lua").live_grep({
              -- cmd = "rg --files",
              winopts = { preview = { hidden = "nohidden" } },
              complete = true 
            })
          end
    , {
        exec_empty_query = true,
    }) end ,  { silent = true, desc = "Fuzzy"     }
    
)
 
```

This is fine, but it does something I don't want, which is to include all this extra stuff in the output. It prints the path of the file to my document, but it also prints the web devicon and the line number corresponding to the line the text shows up in the file. I would really like just the path info. Is there an easy way to do this??

**Reply by ibhagwan** [reply]
Yes, you can either disable icons (undesirable probably) or use a custom complete function and use `return require'fzf-lua'.path.entry_to_file(selected[1]).path` as the return value. I’m on my phone so might not be 100% free of typos, use `vim.print(…)` on any lua object to explore its properties (will show up in `:messages`).

**Reply by ibhagwan** [reply]
Btw, I’m not sure why are you running live grep inside an fzf_exec and not live grep directly? I’m probably missing something about your intent here.

**Comment by radiodee1** [comment]
that works like a charm, except the pasted path replaces all the text in that line, whereas I'd like it to be inserted at the cursor. 

```lua
vim.keymap.set({ "i" }, "<C-x><C-f>",
         function()
            require("fzf-lua").live_grep({ 
              complete = function(selected, opts, line, col)
  
            return require'fzf-lua'.path.entry_to_file(selected[1]).path
        end

            })
          end
    , {}) 

```

**Reply by ibhagwan** [reply]
Correct the return value is a pair that replaces the entire line and sets the cursor (the second return val).

You have the existing line and column as parameters, use `string.xxx` to manipulate the line any ways you wish and return the transformed line instead of just the path, as second value you can use `#line` to set the cursor at the end.

Depending on what you want to achieve here’s how the default complete path behaves:

https://github.com/ibhagwan/fzf-lua/blob/b59bb9a205547a4d57987ab939845d9b35833466/lua/fzf-lua/complete.lua#L67-L78

**Comment by radiodee1** [comment]
thank you.
